lib/pbench/agent/tool_meister.py

```diff
@@ -200,210 +347,141 @@ def stop(self):
 
     def start(self):
-        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
-        if self.pmcd_process or self.pmlogger_process:
+        if not self.pmcd_path:
+            raise RuntimeError("Path to pmcd not provided")
+        if not self.pmlogger_path:
+            raise RuntimeError("Path to pmlogger not provided")
+        if self.pmcd_process is not None:
+            raise ToolException(
+                f"Tool({self.name}) has an unexpected pmcd process running"
+            )
+        if self.pmlogger_process is not None:
             raise ToolException(
-                f"Tool({self.name}) has an unexpected process still running"
+                f"Tool({self.name}) has an unexpected pmlogger process running"
             )
 
+        tool_dir = self.tool_dir / self.real_name
+        self.pmcd_args = [
+            self.pmcd_path,
+            "--foreground",
+            "--socket=./pmcd.socket",
+            "--port=55677",
+            f"--config={self.pbench_install_dir}/templates/pmcd.conf",
+        ]
+        self.pmlogger_args = [
+            self.pmlogger_path,
+            "--log=-",
+            "--report",
+            "-t",
+            "3s",
+            "-c",
+            f"{self.pbench_install_dir}/templates/pmlogger.conf",
+            "--host=localhost:55677",
+            f"{tool_dir}/%Y%m%d.%H.%M",
+        ]
+
         self.logger.info(
-            "%s: start_tool -- %s -- %s",
+            "%s: start_tool -- '%s' && '%s'",
             self.name,
             " ".join(self.pmcd_args),
             " ".join(self.pmlogger_args),
         )
-        o_file = self.tool_dir / f"tm-{self.name}-start.out"
-        e_file = self.tool_dir / f"tm-{self.name}-start.err"
-        with o_file.open("w") as ofp, e_file.open("w") as efp:
-            try:
-                self.pmcd_process = subprocess.Popen(
-                    self.pmcd_args,
-                    cwd=self.tool_dir,
-                    stdin=subprocess.DEVNULL,
-                    stdout=ofp,
-                    stderr=efp,
-                )
-            except Exception as exc:
-                self.logger.error(
-                    "Pmcd run process failed: '%s', %r", exc, self.pmcd_args
-                )
-            try:
-                self.pmlogger_process = subprocess.Popen(
-                    self.pmlogger_args,
-                    cwd=self.tool_dir,
-                    stdin=subprocess.DEVNULL,
-                    stdout=ofp,
-                    stderr=efp,
-                )
-            except Exception as exc:
-                self.logger.error(
-                    "Pmlogger run process failed: '%s', %r", exc, self.pmlogger_args
-                )
+        self.pmcd_process = self._create_process_with_logger(
+            self.pmcd_args, tool_dir, "pmcd"
+        )
+        self.pmlogger_process = self._create_process_with_logger(
+            self.pmlogger_args, tool_dir, "pmlogger"
+        )
```

Comment:
Should we be checking that the processes were created successfully?

Reply:
That'll happen during the `.wait()`.

Reply:
I think (and hope) that that is the wrong answer.  I think the correct answer is, "no, we don't need to check, because any failure would have been reported by an exception."  Am I right?

The problem is, after we return from this function -- and before we call `wait()` -- I think we report back that "all processes were started successfully", so, if we defer the check until we call `wait()`, then I think we end up having lied.